controllers/volsync/vshandler.go

```diff
@@ -700,6 +725,54 @@ func (v *VSHandler) DeleteRD(pvcName string) error {
 	return nil
 }
 
+//nolint:gocognit
+func (v *VSHandler) deleteLocalRDAndRS(rd *volsyncv1alpha1.ReplicationDestination) error {
+	latestRDImage, err := v.getRDLatestImage(rd.GetName())
+	if err != nil {
+		return err
+	}
+
+	v.log.V(1).Info("Latest Image for ReplicationDestination to be used by LocalRS", "name", latestRDImage.Name)
+
+	lrs := &volsyncv1alpha1.ReplicationSource{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      getLocalReplicationName(rd.GetName()),
+			Namespace: v.owner.GetNamespace(),
+		},
+	}
+
+	err = v.client.Get(v.ctx, types.NamespacedName{
+		Name:      lrs.GetName(),
+		Namespace: lrs.GetNamespace(),
+	}, lrs)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			return v.deleteLocalRD(getLocalReplicationName(rd.GetName()))
```

Comment:
Below (line 758) there's a check to make sure the last sync has completed before cleaning up - however if we delete the localRD while that sync is happening, this will prevent it from completing.  It might be cleaner to clean up both localRS and localRD at the same time (after confirming the last sync has completed).

Reply:
We do this only if the localRS is not found.

Reply:
ah, gotcha - my bad